pkg/resolver/framework/testing/reconciler_test.go

```diff
@@ -0,0 +1,177 @@
+/*
+ Copyright 2022 The Tekton Authors
+
+ Licensed under the Apache License, Version 2.0 (the "License");
+ you may not use this file except in compliance with the License.
+ You may obtain a copy of the License at
+
+     http://www.apache.org/licenses/LICENSE-2.0
+
+ Unless required by applicable law or agreed to in writing, software
+ distributed under the License is distributed on an "AS IS" BASIS,
+ WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ See the License for the specific language governing permissions and
+ limitations under the License.
+*/
+
+package testing
+
+import (
+	"encoding/base64"
+	"errors"
+	"testing"
+	"time"
+
+	"github.com/tektoncd/resolution/pkg/apis/resolution/v1alpha1"
+	resolutioncommon "github.com/tektoncd/resolution/pkg/common"
+	"github.com/tektoncd/resolution/test"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	duckv1 "knative.dev/pkg/apis/duck/v1"
+	_ "knative.dev/pkg/system/testing" // Setup system.Namespace()
+)
+
+func TestReconcile(t *testing.T) {
```

Comment:
Ok, so...this is in `pkg/resolver/framework/testing` because `fakecontroller.go` makes the `Resolver` binaries barf with `/ko-app/bundleresolver flag redefined: cluster` if `fakecontroller.go` is in `pkg/resolver/framework` - some Knative testing magic ends up in the resulting `Resolver` binary, calling `ClientConfig.InitFlags(flag.CommandLine)` at init time via, well, magic, resulting in the later call via `sharedmain.MainWithContext` failing. 

So if I wanted to use the fake reconciler framework elsewhere - i.e., in tests for `bundleresolver` etc - I have to have the fake controller/reconciler tooling exported (so not in `_test.go`) but _not_ in the same package as stuff the `Resolver` `main.go`s import. I could just switch this `reconciler_test.go` to not use the fake framework but instead its own copy-pasted equivalent, but...yeah. I'm waiting for someone else to come up with ideas. =)

Reply:
Ok, https://github.com/tektoncd/resolution/pull/51/commits/6721eaece94ebb3a94f9b30fcf689d8e4ac2f7cb seems to fix it, at least at first glance - moving `pkg/clients.go` to `pkg/clients_test.go` got rid of the import chain that led to `knative.dev/pkg/test` ending up in the `Resolver` binaries, and I am pretty sure it won't break anything in the integration tests...

Reply:
Well, it did break something in the integration tests... =)

Reply:
Ok, I gave up, at least for now, and copy-pasted the testing framework code around. Ah well.